fs/fs.go

```diff
@@ -273,9 +273,9 @@ func NewFilesystem(ctx context.Context, root string, cfg config.FSConfig, opts .
 		if err != nil {
 			return nil, fmt.Errorf("cannot create background fetcher: %w", err)
 		}
-		go bgFetcher.Run(context.Background())
+		go bgFetcher.Run(ctx)
```

Comment:
Attached the background fetcher to the context of the filesystem here, this made more sense to me than allowing it to run indefinitely since if the filesystem context gets cancelled, the background fetcher should also cease operations.